cuda/matrix/coo_kernels.cu

```diff
@@ -224,14 +290,30 @@ void spmv2(std::shared_ptr<const CudaExecutor> exec,
 
     auto nwarps = host_kernel::calculate_nwarps(exec, nnz);
     if (nwarps > 0) {
-        int num_lines = ceildiv(nnz, nwarps * cuda_config::warp_size);
-        const dim3 coo_block(cuda_config::warp_size, warps_in_block, 1);
-        const dim3 coo_grid(ceildiv(nwarps, warps_in_block), b->get_size()[1]);
-        abstract_spmv<<<coo_grid, coo_block>>>(
-            nnz, num_lines, as_cuda_type(a->get_const_values()),
-            a->get_const_col_idxs(), as_cuda_type(a->get_const_row_idxs()),
-            as_cuda_type(b->get_const_values()), b->get_stride(),
-            as_cuda_type(c->get_values()), c->get_stride());
+        if (b->get_size()[1] == 1) {
```

Comment:
Seeing the significant performance difference from your last results, I would also include at least the value `2` here. There seems to be something in particular which makes 2 nrhs particularly slow (even slower than one).

Reply:
Yes, I choose 4 for the condition